### DISCUSSION TITLE
sqitch deploy --registry with remote database (oracle)

### DISCUSSION THREAD

**Discussion by joaostorrer** [theory/sqitch#618] [General]
I'm trying out Sqitch with some integration projects and I'm struggling to get dependencies between projects.
Reading issue #313 I've found that registry can be another database, which is exactly what i need.
But this option seems to be not working with oracle database.
My conf file is:

[core]
	engine = oracle
[engine "oracle"]
	target = db:oracle://sqitch:sqitch@/db1
        registry = db:oracle://sqitch:sqitch@/db2

db1 and db2 are included in TNSNAMES.ora file and both sqitch user are created

When I run sqitch deploy it returns:

Adding registry tables to db:oracle://sqitch:sqitch@/db1
CREATE TABLE db:oracle://sqitch:sqitch@/db2.releases (
               *
ERROR at line 1:
ORA-00922: missing or invalid option

And in the db:oracle://sqitch:sqitch@/db2 i already have the sqitch tables

I've tried adding this databases in a target entry (via sqitch target add), but i got the same output

**Comment by theory** [comment]
In Oracle, registries are schemas, not databases. For your second project, use a different registry name and a new schema will be created with that name.

In other words, `registry` should not be a DSN, but just as schema name.

That said, what issue are you having with dependencies. They only work using a single registry. Separate registries have no connections between them.

**Comment by joaostorrer** [comment]
In sqitch Is there a way to deploy in one database and register the changes applied in another database (both oracle databases) ?
I have a project that integrate an ERP with another system, so I have to deploy some objects in the ERP database and other objects in an integration database. In this case, the objects in the integration database are dependent of the objects in ERP database, so I have to have a centralized registry.

**Comment by theory** [comment]
Not in Oracle, no. The idea is that a single database needs to track all the Sqitch projects and their interdependencies, but only for that database. Schemas are ideal for this; the only reason they're not used in other RDBMSes is because they don't have proper schema support.

You cannot depend on objects in different databases. I suggest you either:

1. Don't specify those dependencies. Consider them separate projects for separate databases
2. Deploy the ERP project changes to the integration database using `--log-only`. That will record the changes in the registry without actually deploying them.

I recommend option 1. It's really best to think of dependencies only on a per-database bases, not between databases.
